cpp/stdlib.cc

```diff
@@ -137,12 +152,14 @@ time_t time() {
 // 32 bits.  Point being, using anything but the time_t typedef here could
 // (unlikely, but possible) produce weird behavior.
 time_t localtime(time_t ts) {
+  NO_ROOTS_FRAME(FUNC_NAME);  // No allocations here
   tm* loc_time = ::localtime(&ts);
   time_t result = mktime(loc_time);
   return result;
 }
 
 Str* strftime(Str* s, time_t ts) {
+  RootsFrame _r{FUNC_NAME};
```

Comment:
I would call the use of `RootsFrame` here an optimization for the `e_die()` case. The exception handler will immediately free the string allocated here if it happens to alloc. Probably not necessary and happy to switch to `NO_ROOTS_FRAME` if that would be easier to understand